src/Bicep.Core/Analyzers/Linter/Rules/BCPL1000.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1000 : LinterRule
+    {
+        public BCPL1000() : base(
+            code: "BCPL1000",
+            ruleName: "Parameters Required",
+            description: "A valid template must be parameterized.",
+            enableForCLI: true,
+            enableForEdit: true,
+            diagnosticLevel: Diagnostics.DiagnosticLevel.Error,
+            docUri: "https://bicep/linter/rules/BCPL1000") //TODO: set up online documentation location
+        {
+        }
+
+        override public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            bool parametersExist = model.Root.ParameterDeclarations.Any();
+
+            if (!parametersExist)
+            {
+                //TODO: what span is appropriate for a document level diagnostic
```

Comment:
>TODO [](http://example.com/codeflow?start=18&length=4)

I don't know the answer, but here are some thoughts:
1. Could check what the copyright header linter in the Bicep repo does. The concept is similar over there.
2. Could just pick the span of the first Token we see in `ProgramSyntax` and then fall back to [0,0) if the file is empty.

Reply:
I feel like `[0,0)` is most appropriate here personally.

Reply:
Changed this to 0 position with length of 0.